src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1264,22 +1264,14 @@ static void fill_idPtypeNP
   const std::string&  typePatternQ
 )
 {
-  std::string typeCheck;
-
-  // Sometimes (for instance, when we reach this point during a entity deletion operation)
-  // we don't know the type of the entity, in which case the check for type shouldn't be included
-  if (!entityType.empty())
-  {
-    typeCheck = std::string("(this.")+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \""+entityType+"\" || " +
-        "this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \"\" || " +
-        "!(\""+CSUB_ENTITY_TYPE+"\" in this."+CSUB_ENTITIES+"[i])) && ";
-  }
-
   bgP->functionIdPtypeNP = std::string("function()") +
          "{" +
             "for (var i=0; i < this."+CSUB_ENTITIES+".length; i++) {" +
                 "if (this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_ISPATTERN+" == \"true\" && " +
-                    "!this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_ISTYPEPATTERN+" && " + typeCheck +
+                    "!this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_ISTYPEPATTERN+" && " +
+                    "(this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \""+entityType+"\" || " +
+                        "this."+CSUB_ENTITIES+"[i]."+CSUB_ENTITY_TYPE+" == \"\" || " +
+                        "!(\""+CSUB_ENTITY_TYPE+"\" in this."+CSUB_ENTITIES+"[i])) && " +
```

Comment:
This is the rollback of a change introduced in recent PR https://github.com/telefonicaid/fiware-orion/pull/4084. It was introduced to solve the entityDelete case with empty type.

However, this is not needed taking into account the changes in updateEntity() introduced in this PR. Probably it doesn't hurt to keep as it was left in PR #4084. Howver, by the law of minimum intervention :), let's leave a it was before.

NTC (informative)